dimager/views.py

```diff
@@ -1,3 +1,109 @@
 from django.shortcuts import render
+from django.contrib.auth.decorators import login_required
+from dimager.models import ImagerProfile
+from imager_images.models import Album, Photo, User
+from django.views.generic import CreateView, UpdateView, DeleteView
 
-# Create your views here.
+
+@login_required()
+def user_profile(request, id):
+    user = User.objects.get(id=id)
+    try:
+        albums = Album.objects.filter(user=user).all()
+    except:
+        albums = None
+    profile = ImagerProfile.objects.get(user=user)
+    num_photos = Photo.objects.filter(user=user).count()
+    return render(request, 'profile.html', {
+        'albums': albums,
+        'user': user,
+        'profile': profile,
+        'num_photo': num_photos
+    })
+
+
+@login_required()
+def user_profile(request):
+    user = request.user
+    try:
+        albums = Album.objects.filter(user=user).all()
+    except:
+        albums = None
+    profile = ImagerProfile.objects.get(user=user)
+    num_photos = Photo.objects.filter(user=user).count()
+    return render(request, 'profile.html', {
+        'albums': albums,
+        'user': user,
+        'profile': profile,
+        'num_photo': num_photos
+    })
```

Comment:
This is odd.  I swear you have exactly this same function defined on line 9 above.  Why the same thing twice?  Does it actually work for you? (I don't think it does.)